tests/terminal.rs

```diff
@@ -47,3 +50,96 @@ fn terminal_draw_returns_the_completed_frame() -> Result<(), Box<dyn Error>> {
     assert_eq!(frame.area, Rect::new(0, 0, 8, 8));
     Ok(())
 }
+
+#[test]
+fn terminal_insert_before_moves_viewport() -> Result<(), Box<dyn Error>> {
```

Comment:
As it would only be the y position that is changed I thought it clearer to use the visual assertion of the buffer content, I think this is now more clear with the recent test improvements but if you still wish I can do this too.